sybil/sybil.py

```diff
@@ -92,8 +93,10 @@ def __init__(
     ):
 
         self.parsers: Sequence[Parser] = parsers
-        calling_filename = sys._getframe(1).f_globals.get('__file__')
-        if calling_filename:
+        current_frame = inspect.currentframe()
+        if current_frame is not None:
+            calling_frame = current_frame.f_back
+            calling_filename = inspect.getframeinfo(calling_frame).filename
```

Comment:
Is the `else` clause below still needed? Coverage says no...

Reply:
Yes: I do 100% coverage, hence the CI failure.

No idea which test, I'd checkout master and must a raise in the branch to see

Reply:
I looked into this, and yeah, the else block can go. Inspect is still able to get the calling filename even when `__file__` is not found in the frame globals, since it uses a different mechanism (it looks at the importlib loader detail). Using importlib loader to discover the caller is both more direct and more correct, since importlib itself is what sets the `__file__` attribute after importing, so you cut out the middleman this way.

As for why the else block was needed previously, I think it may because of your `ClearNamespaceParser`? The files it failed to inspect the calling filename from were dogfooding examples in .rst which demonstrated the `ClearNamespaceParser` usage. I suppose it kicked `__file__` out of the namespace, which was causing the lookup of `"__file__"` in the frame globals to fail.